sklearnex/_device_offload.py

```diff
@@ -200,9 +201,25 @@ def _copy_to_usm(queue, array):
         raise RuntimeError(
             "dpctl need to be installed to work " "with __sycl_usm_array_interface__"
         )
-    mem = MemoryUSMDevice(array.nbytes, queue=queue)
-    mem.copy_from_host(array.tobytes())
-    return usm_ndarray(array.shape, array.dtype, buffer=mem)
+    if hasattr(array, "nbytes"):
+        mem = MemoryUSMDevice(array.nbytes, queue=queue)
+        mem.copy_from_host(array.tobytes())
+        return usm_ndarray(array.shape, array.dtype, buffer=mem)
+    else:
+        if isinstance(array, Iterable):
+            array = [_copy_to_usm(queue, i) for i in array]
+        return array
```

Comment:
I would like to avoid this recursion here. It seems to me more safe

```suggestion
    def copy_to_usm(queue, array): #better name is needed
        mem = MemoryUSMDevice(array.nbytes, queue=queue)
        mem.copy_from_host(array.tobytes())
        return usm_ndarray(array.shape, array.dtype, buffer=mem)
   
    if hasattr(array, "nbytes"):
        return copy_to_usm(queue, array)
    else:
        if isinstance(array, Iterable):
            array = [copy_to_usm(queue, i) for i in array]
        return array
```